plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -64,21 +61,12 @@
     private final BigQuery bigQuery;
     private final ViewMaterializationCache materializationCache;
     private final boolean caseInsensitiveNameMatching;
-    private final Cache<String, Optional<RemoteDatabaseObject>> remoteDatasets;
-    private final Cache<TableId, Optional<RemoteDatabaseObject>> remoteTables;
```

Comment:
cc @hashhar see `Remove unused caches in BigQueryClient`.

Reply:
🤦‍♂️ 

Indeed. It has been present since the initial commit. I compute a fresh mapping near `// explicitly cache the information if the requested dataset doesn't exist` but don't put it into the cache at all.

I'd prefer fixing this instead of removing it. The `@DefunctConfig` with a TODO should be good for now.

Reply:
> I'd prefer fixing this instead of removing it though (in separate PR).

understood, except that i didn't want to spend time fixing this particular one.

i can drop the change. note that this is not the only problem, probably -- what about cache invalidation?

Reply:
`Remove unused caches in BigQueryClient` dropped

Reply:
@grantatspothero is both fixing this to actually populate the cache and converting to use the EvictableCache to prevent leaking keys that are already being loaded.

Note that we don't have a procedure to clear the entire cache so it's a better situation than the JDBC connectors.

Reply:
Let's remove the cache for now. We ran into some isssues - bulk loading not playing well with loading caches, no invalidation logic when we make changes via Trino etc.

@findepi Can you restore your commit? Sorry for the back and forth.

Reply:
> We ran into some isssues - bulk loading not playing well with loading caches, no invalidation logic when we make changes via Trino etc.

@hashhar @grantatspothero 
bulk list + `cache.put` is inherently not playing well with invalidation

in JDBC, we 'solved' this problem by not having invalidation (`CachingIdentifierMapping`)
cc @kokosing

Reply:
> Can you restore your commit?

https://github.com/trinodb/trino/pull/10725